django/views/generic/list.py

```diff
@@ -59,7 +59,7 @@ def paginate_queryset(self, queryset, page_size):
         page_kwarg = self.page_kwarg
         page = self.kwargs.get(page_kwarg) or self.request.GET.get(page_kwarg) or 1
         try:
-            page_number = int(page)
+            page_number = paginator.validate_number(page)
```

Comment:
This would no longer raise `ValueError` but `PageNotAnInteger` or `EmptyPage`. I think this PR needs some more consideration and tests.